omod/src/main/java/org/openmrs/web/attribute/handler/RegexValidatedTextDatatypeHandler.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.web.attribute.handler;
+
+import java.util.Map;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openmrs.api.context.Context;
+import org.openmrs.customdatatype.CustomDatatype;
+import org.openmrs.customdatatype.InvalidCustomValueException;
+import org.openmrs.customdatatype.datatype.RegexValidatedTextDatatype;
+import org.springframework.stereotype.Component;
+
+/**
+ * A handler for the {@link RegexValidatedTextDatatype}.
+ */
+@Component
+public class RegexValidatedTextDatatypeHandler implements FieldGenDatatypeHandler<RegexValidatedTextDatatype, java.lang.String> {
+	
+	
+	private static final Log log = LogFactory.getLog(RegexValidatedTextDatatypeHandler.class);
+	
+	/**
+	 * @see org.openmrs.customdatatype.CustomDatatypeHandler#setHandlerConfiguration(String)
+	 */
+	@Override
+	public void setHandlerConfiguration(String arg0) {
+		// not used
+	}
+	
+	/**
+	 * @see org.openmrs.web.attribute.handler.FieldGenDatatypeHandler#getWidgetName()
+	 */
+	@Override
+	public String getWidgetName() {
+		return "java.lang.String";
```

Comment:
@wluyima if I do as you say, I get

![image](https://cloud.githubusercontent.com/assets/4661144/16520730/67553d2e-3f92-11e6-9921-b37cf6346660.png)

because there is no such type. Looking for example to the [ProviderFieldGenDatatypeHandler ](ProviderFieldGenDatatypeHandler) it is "org.openmrs.Provider" and there is a fieldgen tag for the provider.